grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java

```diff
@@ -320,8 +314,7 @@ void shutdown() {
   }
 
   void propagateError(Status status) {
-    logger.log(Level.FINE, "[{0}] Had an error: {1}; dropList={2}; backendList={3}",
-        new Object[] {logId, status, dropList, backendList});
+    logger.log(Level.DEBUG, "Error: " + status);
```

Comment:
We really want to know whether or not this log message will be used, to avoid the string handling when useless.

Reply:
Since ChannelLogger is not on data path and is called pretty infrequently, I am not sure it's worth the code clutter to wrap it with a check.

Reply:
Things like `logger.log(Level.DEBUG, "Got an LB response: " + response);` could still be large and run on the network thread. Also, some LBs may do RPCs on cache misses, which could want to log to DEBUG.

I'm a bit worried about how much is getting pulled into this API. There's a lot of data being logged. Like every time the picker changes with `logger.log(Level.INFO, state + ": picks=" + picker.pickList + ", drops=" + picker.dropList);`. I know that isn't on the critical path, but that still happens frequently. And most of the time nobody will see it.

Reply:
What about `log(Level level, String template, Object... args)`?. For now I can check the loggable level in the implementation and that would save the string generation for `DEBUG`. We could optimize `ChannelTracer` even later to generate strings only when Channelz is requested.

Reply:
`log(Level, String, Object...)` seems fine to me. It'd probably be an overload (so `log(Level, String)` would still exist) to avoid allocating the `Object[]` when unnecessary.